api/utils/requestProcessor.js

```diff
@@ -89,6 +89,20 @@ const processRequest = (params) => {
     if (!params.req || !params.req.url) {
         return common.returnMessage(params, 400, "Please provide request data");
     }
+
+    if (params.res) {
+        //PROXY THE RESPONSE WRITE FUNCTION HERE TO SET RESPONSE OBJECT TO PARAMS
+        var write = params.res.write.bind(params.res);
+        params.res.write = function(data, encoding, callback) {
```

Comment:
There are actually cases where we do not write response, but return response in a callback. Like using requestProcessor programmatically. Maybe it makes more sense rather than overwriting write function, to better cover 3 response methods that we have here:
https://github.com/Countly/countly-server/blob/next/api/utils/common.js#L974-L1132
to store response in params as object. And not messing with nodejs internals too :)